e2e/cypress/support/command.js

```diff
@@ -18,6 +18,21 @@ Cypress.Commands.add('check_harvest_done', (retries) => {
         });
 });
 
+Cypress.Commands.add('check_dataset_harvested', (retries) => {
+    // harvester is considered functioning if we get at least 1 dataset in. 
+    cy.reload(true)
+
+    cy.contains('#content section.module-content dl dt', 'Datasets').next().invoke('text').then($text => {
+        if ($text == '0') {
+            cy.wait(1000);
+            cy.check_dataset_harvested(retries - 1);
```

Comment:
If I'm reading this right, you need to put the `retries == 0` check first... 
Otherwise, when `$text == 0` & `retries == 0`, the second block will never get tripped.

Reply:
Good catch.